File: CombSort.java
package Sort;

import java.util.Scanner;

public class CombSortHW {
    public static final int MAX_SIZE = 1000;

    public static void main(String[] args){
        int numTestCases;
        Scanner sc = new Scanner(System.in);
        numTestCases = sc.nextInt();

        for (int i = 0; i < numTestCases; ++i) {
            int num;
            int[] a = new int[MAX_SIZE];

            num = sc.nextInt();
            for (int j = 0; j < num; j++) {
                a[j] = sc.nextInt();
            }
            combSort(a, num);
        }
    }

    private static void swap(int a[], int i, int j){
        int tmp = a[i];
        a[i] = a[j];
        a[j] = tmp;
    }

    public static void combSort(int a[], int n){
        int countCmpOps = 0;
        int countSwaps = 0;

        double shrink = 1.3;
        int gap = n;
        boolean sorted = false;

        while(sorted == false){
            gap = (int)(gap/shrink);
            if(gap<=1){
                gap = 1;
                sorted = true;
            }
            int i =0;
            while(i+gap < n){
                if(a[i] > a[i+gap]){
                    swap(a,i, i+gap );
                    sorted = false;
                    countSwaps++;
                }
                i++;
                countCmpOps++;
            }
        }
        System.out.printf("%d %d \n", countCmpOps, countSwaps);
    }
}

File: Cpp/Sort/CombSort.cpp
#include <iostream>

#define MAX_SIZE 1000

using namespace std;

void combSort(int arr[], int n);
void swap(int* a, int* b);

int main(){
    int numTestCase;

    cin >> numTestCase;

    for(int i = 0; i < numTestCase; i++){
        int n;
        cin >> n;
        int num[MAX_SIZE];
        for(int j = 0; j<n; j++){
            cin >> num[j];
        }
        combSort(num, n);
    }
    return 0;
}

void combSort(int arr[], int n){
    double k = 1.3;
    int gap = n/k;
    while(gap >= 1){
        for(int i = 0; i<n-gap; i++){
            if(arr[i] > arr[i+gap]){
                swap(&arr[i], &arr[i+gap]);
            }
        }
        gap /= k;
    }
    for(int i = 0; i<n; i++){
        cout << arr[i] << " ";
    }
    cout << endl;
}

void swap(int* a, int* b){
    int tmp = *a;
    *a = *b;
    *b = tmp;
}

